services/motion/builtin/builtin.go

```diff
@@ -79,31 +83,35 @@ func (ms *builtIn) Reconfigure(
 	ms.lock.Lock()
 	defer ms.lock.Unlock()
 
-	slamServices := make(map[resource.Name]slam.Service)
+	localizers := make(map[resource.Name]motion.Localizer)
 	components := make(map[resource.Name]resource.Resource)
 	for name, dep := range deps {
 		switch dep := dep.(type) {
 		case framesystem.Service:
 			ms.fsService = dep
-		case slam.Service:
-			slamServices[name] = dep
+		case slam.Service, movementsensor.MovementSensor:
```

Comment:
I don't think this is a huge deal for 2 types.  If it grows to a third we should think about a more sustainable way to do this.  Reason I say this is because it isn't obvious to me how you're supposed to use a list as a case in a switch statement

Reply:
hmmm, im not sure if
```
case isLocalizer(dep):
```
is possible.
error: `expected ':', found '('syntax`

Reply:
I think we'll build up the infastructure around these as time goes on.  Hypothetically we could actually get the components to natively implement this method which would make things really clean

Reply:
>Hypothetically we could actually get the components to natively implement this interface which would make things really clean

that would require a scope though, right?